pkg/clients/zone/zone.go

```diff
@@ -0,0 +1,101 @@
+package zone
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/route53iface"
+	"github.com/google/uuid"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+)
+
+// Client defines Route53 Client operations
+type Client interface {
+	CreateZoneRequest(cr *v1alpha3.Zone) route53.CreateHostedZoneRequest
+	DeleteZoneRequest(id *string) route53.DeleteHostedZoneRequest
+	GetZoneRequest(id string) route53.GetHostedZoneRequest
+	UpdateZoneRequest(id, comment *string) route53.UpdateHostedZoneCommentRequest
+}
+
+type zoneClient struct {
+	zone route53iface.ClientAPI
+}
+
+// NewClient creates new AWS Client with provided AWS Configurations/Credentials
+func NewClient(config *aws.Config) Client {
+	return &zoneClient{zone: route53.New(*config)}
+}
+
+// GetZoneRequest returns a route53 GetHostedZoneRequest using which a route53
+// Hosted Zone can be fetched and checked for existence.
+func (c *zoneClient) GetZoneRequest(id string) route53.GetHostedZoneRequest {
+	return c.zone.GetHostedZoneRequest(&route53.GetHostedZoneInput{
+		Id: &id,
+	})
+}
+
+// CreateZoneRequest returns a route53 CreateHostedZoneRequest using which a route53
+// Hosted Zone can be created.
+func (c *zoneClient) CreateZoneRequest(cr *v1alpha3.Zone) route53.CreateHostedZoneRequest {
```

Comment:
Same as above comment. The preference has been that the actual AWS API call happens in controller code. Client code is usually used for producing inputs for those calls and converting their output to be used in our structs.

Reply:
This has been changed accordingly, as suggested.